lib/ansible/module_utils/urls.py

```diff
@@ -300,6 +302,51 @@ def http_open(self, req):
         return self.do_open(UnixHTTPConnection(self._unix_socket), req)
 
 
+class ProxyHandler(urllib.request.ProxyHandler):
+    _SPLITPORT_RE = re.compile('(.*):([0-9]*)', re.DOTALL)
```

Comment:
Given the maximum port number, I'd restrict this to 5 digits...
```suggestion
    _SPLITPORT_RE = re.compile('(.*):([0-9]{1,5})', re.DOTALL)
```